### ISSUE TITLE
Uncaught TypeError: Cannot read property 'name' of null [website_sign model]

### ISSUE THREAD

**Issue by dan18fan** [odoo/odoo#18973]
Impacted versions: 10.0e

Steps to reproduce: In the eSignature app: Uploaded pdf, sending worked, reinstalled (to delete the testing stuff i did in the app), started to fill database with contacts and stuff, after a while the error appeared out of nowhere.

Current behavior: When I click "Send" to request someone to sign any document it gives me this error:

`Uncaught TypeError: Cannot read property 'name' of null
http://192.168.10.199:8069/web/content/1284-4174716/web.assets_backend.js:2953
Rastreo de error:
TypeError: Cannot read property 'name' of null
    at Object.prepare_reference (http://192.168.10.199:8069/web/content/1284-4174716/web.assets_backend.js:2953:1706)
    at Object.<anonymous> (http://192.168.10.199:8069/web/content/1265-156d856/web.assets_common.js:547:681)
    at fire (http://192.168.10.199:8069/web/content/1265-156d856/web.assets_common.js:541:299)
    at Object.fireWith [as resolveWith] (http://192.168.10.199:8069/web/content/1265-156d856/web.assets_common.js:546:198)
    at Object.<anonymous> (http://192.168.10.199:8069/web/content/1265-156d856/web.assets_common.js:547:874)
    at fire (http://192.168.10.199:8069/web/content/1265-156d856/web.assets_common.js:541:299)
    at Object.fireWith [as resolveWith] (http://192.168.10.199:8069/web/content/1265-156d856/web.assets_common.js:546:198)
    at Object.<anonymous> (http://192.168.10.199:8069/web/content/1265-156d856/web.assets_common.js:547:874)
    at fire (http://192.168.10.199:8069/web/content/1265-156d856/web.assets_common.js:541:299)
    at Object.fireWith [as resolveWith] (http://192.168.10.199:8069/web/content/1265-156d856/web.assets_common.js:546:198)`

I have been analyzing the web.assets_backend code at the function "prepare_reference" and apparently its trying to "query" or get info from the many2one variable "name" from the model res.partner

**ResPartners.query(['name']).filter([['id','=',user.partner_id[0]]]).first().then(prepare_reference);}));function prepare_reference(partner){self.default_reference="-";var split=partner.name.split(' ');for(var i=0;i<split.length;i++){self.default_reference+=split[i][0];}}},start:function()**

Expected behavior: Its supposed to open a popup form/window where it asks me to write whos the receiver to sign and whos the sender.

Video/Screenshot link (optional):
![image](https://user-images.githubusercontent.com/29781986/29528332-45897586-8661-11e7-9627-0590919ab73c.png)

**Comment by Yenthe666** [comment]
@ged-odoo the error itself seems JS related atleast. Could you have a look?

**Comment by ged-odoo** [comment]
@qsm-odoo looks like the code expects a res.partner, but the query returns nothing. I'll let qsm have a look at this one

**Comment by dan18fan** [comment]
Thanks for the replies guys, I did something to see if its a problem with the local installation or the database, I installed odoo enterprise in my personal laptop cleanly, I did a fresh new database and installed this app, worked as intended, then I went to databases manager, restored a database with the database I have in my job's local server (the one that throws me this error) and boom... the same error, to make sure it wasn't a code problem or installation problem I went to the new database and it was still working, so I assume it has something to do with the database, in the damaged database I started to delete all the apps and contacts to just leave it as a "fresh database", then I installed this app again and it  is still giving me the same error.

All I remember that I did with this app when I installed it, I uploaded a PDF, sent it to my boss, then I uninstalled the app because I saw it was working, so I wanted the data to be gone from that app and do a new fresh install of that app, I don't really remember testing it again after the reinstall, but after a while of filling sales orders, filling contacts, employees, creating users, when I wanted to test this app again, it suddenly gave me this error.

I don't really want to reinstall again and fill everything again (I already id it once) its exhausting, is there a way to fix this issue without touching the data in other apps?

In advance of anyhelp incoming, thank you very much!
